booking-app/src/main/java/org/motechproject/bookingapp/service/impl/VisitScheduleServiceImpl.java

```diff
@@ -115,6 +157,25 @@ public void savePlannedVisitDates(String subjectId, LocalDate primeVaccinationDa
         visitBookingDetailsService.createOrUpdate(calculatePlannedDates(subject.getVisits(), primeVaccinationDate).values());
     }
 
+    private void validateDate(String subjectId, LocalDate date) {
+        LocalDate screeningDate = visitDataService.findVisitBySubjectIdAndType(subjectId, VisitType.SCREENING).getDate();
+
+        LocalDate earliestDate;
+        LocalDate latestDate;
+
+        if (isFemaleChildBearingAge(subjectId)) {
```

Comment:
in calculatePlannedDates you will have details for prime vac, so request in is not needed also